src/NuGet.Core/NuGet.Common/IObjectWriter.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGet.Common
+{
```

Comment:
Agree with @jainaashish. The scope of this interface should be narrowed to the library it's actually used. 
Also as its name suggests it's supposed to be generic purpose entity but it's far away from being generic.